frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -599,8 +599,9 @@ final class BloopBspServices(
         projects.iterator.map {
           case (target, project) =>
             val sources = project.sources.map { s =>
+              val kind = if(s.isDirectory) bsp.SourceItemKind.Directory else bsp.SourceItemKind.File
```

Comment:
`sources` can contain both source dirs and source files. if a path doesn't exist, `isDirectory` will be false and the kind will be file (even though it could be a non-existing directory). To fix this issue, we should imitate what we do in `toBspSourceUri`: if a path doesn't exist, use file item kind only if the file name looks like a source file name, otherwise default on a directory item kind. https://github.com/scalacenter/bloop/blob/de71440426b75e50106053d354acda6993d4299e/backend/src/main/scala/bloop/io/AbsolutePath.scala#L27-L36

Reply:
I added an optional parameter to `isFile` - does this make sense?